modules/firehose/module.go

```diff
@@ -0,0 +1,122 @@
+package firehose
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/odpf/entropy/domain"
+	gjs "github.com/xeipuuv/gojsonschema"
+	"go.uber.org/zap"
+)
+
+const configSchemaString = `
+{
+	"$schema": "http://json-schema.org/draft-04/schema#",
+	"type": "object",
+	"properties": {
+		"image": { "type": "string" },
+		"configuration": {
+			"type": "object",
+			"properties": {
+				"SOURCE_KAFKA_BROKERS": { "type": "string" },
+				"SOURCE_KAFKA_TOPIC": { "type": "string" },
+				"SOURCE_KAFKA_CONSUMER_GROUP_ID": { "type": "string" },
+				"INPUT_SCHEMA_PROTO_CLASS": { "type": "string" },
+			}
+		},
+		"replicas": { "type": "number" },
+		"namespace": { "type": "string" },
+		"cluster": { "type": "string" },
+		"sink_type": { "type": "string",},
+		"state": { "type": "string", "enum": [ "RUNNING", "STOPPED" ] },
+		"stop_date": { "type": "string", "format": "date-time" },
+		"description": { "type": "string" },
+	},
+	"oneOf": [
+		{
+		  "if": {
+			"properties": { "sink_type": { "const": "LOG" } }
+		  },
+		  "then": {
+			"properties": { "configuration": { 
+				"KAFKA_RECORD_PARSER_MODE": { "type": "string" },
+				"SOURCE_KAFKA_BROKERS": { "type": "string" },
+				"SOURCE_KAFKA_TOPIC": { "type": "string" },
+				"SOURCE_KAFKA_CONSUMER_GROUP_ID": { "type": "string" },
+				"INPUT_SCHEMA_PROTO_CLASS": { "type": "string" },
```

Comment:
These appear above as well at lines 22-25. Let's keep it only at one place.

Reply:
Will remove it from lines 22-25. The configuration will be added based on the `sink_type`, in the `oneOf` conditional validation.